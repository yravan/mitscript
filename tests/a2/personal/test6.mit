// Expressions test 

// Logical Operators
print("True and False: " + (true & false));
print("True or False: " + (true | false));
print(" ");

// Arithmetic Operators
print("6 + 3: " + (6 + 3));
print("6 - 3: " + (6 - 3));
print("6 * 3: " + (6 * 3));
print("6 / 3: " + (6 / 3));
print("6 + \"3\": " + (6 + "3"));
print(" ");

// Unary Operators
print("-3: " + (-3));
print("Not True: " + (!true));
print(" ");

// Comparison Operators
print("1 > 0: " + (1 > 0));
print("1 < 0: " + (1 < 0));
print("1 >= 0: " + (1 >= 0));
print("1 <= 0: " + (1 <= 0));
print(" ");
print("None == None: " + (None == None));
print("1 == 1: " + (1 == 1));
print("1 == 0: " + (1 == 0));
print("1 == intcast(\"1\"): " + (1 == intcast("1")));
print("\"1\" == \"1\": " + ("1" == "1"));
print("\"1\" == \"0\": " + ("1" == "0"));
print("true == true: " + (true == true));
print("false == true: " + (false == true));
a = {z:2;x:1;};
b = {z:2;x:1;};
c = a;
print("a == b: " + (a == b));
print("a == c: " + (a == c));

x = fun(a,b){print(a+b);};
y = fun(a,b){return(a+b);};
z = x;
print("x == y: " + (x == y));
print("x == z: " + (x == z));

print("a == x: " + (a == x));
print("x == 10: " + (x == 10));

//Variable Read + FieldRead + IndexRead
print("a: " + a);
print("a.x: " + a.x);
print("a.y: " + a.y);
tmp = "z";
print("a[\"z\"]: " + a[tmp]);

