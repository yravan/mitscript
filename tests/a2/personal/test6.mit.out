True and False: false
True or False: true
 
6 + 3: 9
6 - 3: 3
6 * 3: 18
6 / 3: 2
6 + "3": 63
 
-3: -3
Not True: false
 
1 > 0: true
1 < 0: false
1 >= 0: true
1 <= 0: false
 
None == None: true
1 == 1: true
1 == 0: false
1 == intcast("1"): true
"1" == "1": true
"1" == "0": false
true == true: true
false == true: false
a == b: false
a == c: true
x == y: false
x == z: true
a == x: false
x == 10: false
a: {x:1 z:2 }
a.x: 1
a.y: None
a["z"]: 2
